### ISSUE TITLE
Restart device-virtual failed with runtime error

### ISSUE THREAD

**Issue by cherrycl** [edgexfoundry/device-virtual-go#195]
### Reproduce Steps
1. Download the [docker-compose file](https://github.com/edgexfoundry/edgex-compose/blob/d11d4919902da1f322d0dd8f966d29acaea177d2/taf/docker-compose-taf-perf.yml) from edgex-compose.
2. Start device-virtual with command `docker-compose -f docker-compose-taf-perf-no-secty.yml up -d device-virtual`.
3. Restart device-virtual with command `docker-compose -f docker-compose-taf-perf-no-secty.yml restart device-virtual`

### Error
Restart service failed and got an error in the docker log.
```
level=INFO ts=2021-05-04T10:06:42.975831365Z app=device-virtual source=profiles.go:73 msg="Profile Random-UnsignedInteger-Device exists, using the existing one"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x8b7816]

goroutine 1 [running]:
github.com/edgexfoundry/device-sdk-go/v2/internal/provision.LoadDevices(0xc0003660b0, 0xd, 0xc0002a9360, 0x0, 0x0)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/internal/provision/devices.go:48 +0x1d6
github.com/edgexfoundry/device-sdk-go/v2/pkg/service.(*Bootstrap).BootstrapHandler(0xc00021fcc0, 0xc6e2e0, 0xc0002aca00, 0xc000286b10, 0xc01c650cb69b2b2b, 0x8ee91c, 0xfef5c0, 0xdf8475800, 0x3b9aca00, 0xc0002a9360, ...)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/pkg/service/init.go:74 +0x43e
github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap.RunAndReturnWaitGroup(0xc6e2e0, 0xc0002aca00, 0xc000296d90, 0xc743e0, 0xc0002b0420, 0xba0a4c, 0xe, 0xba0a84, 0xe, 0xc73540, ...)
	/go/pkg/mod/github.com/edgexfoundry/go-mod-bootstrap/v2@v2.0.0-dev.34/bootstrap/bootstrap.go:158 +0x59d
github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap.Run(0xc6e2e0, 0xc0002aca00, 0xc000296d90, 0xc743e0, 0xc0002b0420, 0xba0a4c, 0xe, 0xba0a84, 0xe, 0xc73540, ...)
	/go/pkg/mod/github.com/edgexfoundry/go-mod-bootstrap/v2@v2.0.0-dev.34/bootstrap/bootstrap.go:184 +0x158
github.com/edgexfoundry/device-sdk-go/v2/pkg/service.Main(0xba0a4c, 0xe, 0xc4ff10, 0xc, 0xb3cf60, 0xc0000694a0, 0xc6e2e0, 0xc0002aca00, 0xc000296d90, 0xc0002b6540)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/pkg/service/main.go:65 +0x7fb
github.com/edgexfoundry/device-sdk-go/v2/pkg/startup.Bootstrap(0xba0a4c, 0xe, 0xc4ff10, 0xc, 0xb3cf60, 0xc0000694a0)
	/go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/pkg/startup/bootstrap.go:19 +0x10e
main.main()
	/device-virtual-go/cmd/main.go:22 +0x72
```

**Comment by cherrycl** [comment]
The full log is as below, after changing the loglevel to `DEBUG`.
```
level=INFO ts=2021-05-05T03:33:34.7765335Z app=device-virtual source=restrouter.go:48 msg="Registering v2 routes..."
level=DEBUG ts=2021-05-05T03:33:34.78680148Z app=device-virtual source=service.go:194 msg="trying to find device service device-virtual"
level=INFO ts=2021-05-05T03:33:34.791986172Z app=device-virtual source=service.go:212 msg="device service device-virtual exists, updating it"
2021/05/05 03:33:34 http: panic serving 172.18.0.5:58564: runtime error: invalid memory address or nil pointer dereference
goroutine 106 [running]:
net/http.(*conn).serve.func1(0xc0004b4320)
 /usr/local/go/src/net/http/server.go:1801 +0x147
panic(0xae2880, 0xf8dd30)
 /usr/local/go/src/runtime/panic.go:975 +0x47a
github.com/edgexfoundry/device-sdk-go/v2/internal/application.UpdateDeviceService(0xc000460ef0, 0x2, 0xc0004383f0, 0x24, 0xc000488300, 0xc000488310, 0xc000488320, 0x0, 0x0, 0x0, ...)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/internal/application/callback.go:228 +0x140
github.com/edgexfoundry/device-sdk-go/v2/internal/controller/http.(*RestController).UpdateDeviceService(0xc0004084b0, 0xc6cae0, 0xc0000d00e0, 0xc00044e600)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/internal/controller/http/callback.go:166 +0x1bd
net/http.HandlerFunc.ServeHTTP(0xc0004186f0, 0xc6cae0, 0xc0000d00e0, 0xc00044e600)
 /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/edgexfoundry/device-sdk-go/v2/internal/controller/http/correlation.LoggingMiddleware.func1.1(0xc6cae0, 0xc0000d00e0, 0xc00044e600)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/internal/controller/http/correlation/middleware.go:42 +0xb3
net/http.HandlerFunc.ServeHTTP(0xc0004927b0, 0xc6cae0, 0xc0000d00e0, 0xc00044e600)
 /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/edgexfoundry/device-sdk-go/v2/internal/controller/http/correlation.ManageHeader.func1(0xc6cae0, 0xc0000d00e0, 0xc00044e500)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/internal/controller/http/correlation/middleware.go:28 +0x1f1
net/http.HandlerFunc.ServeHTTP(0xc0004903c0, 0xc6cae0, 0xc0000d00e0, 0xc00044e500)
 /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/edgexfoundry/device-sdk-go/v2/internal/controller/http/correlation.RequestLimitMiddleware.func1.1(0xc6cae0, 0xc0000d00e0, 0xc00044e500)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/internal/controller/http/correlation/middleware.go:54 +0x5a
net/http.HandlerFunc.ServeHTTP(0xc0004903e0, 0xc6cae0, 0xc0000d00e0, 0xc00044e500)
 /usr/local/go/src/net/http/server.go:2042 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0002b4540, 0xc6cae0, 0xc0000d00e0, 0xc00044e300)
 /go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0xd3
net/http.serverHandler.ServeHTTP(0xc0000d0700, 0xc6cae0, 0xc0000d00e0, 0xc00044e300)
 /usr/local/go/src/net/http/server.go:2843 +0xa3
net/http.(*conn).serve(0xc0004b4320, 0xc6e2e0, 0xc00049a180)
 /usr/local/go/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
 /usr/local/go/src/net/http/server.go:2969 +0x36c
level=INFO ts=2021-05-05T03:33:34.799261232Z app=device-virtual source=profiles.go:51 msg="Loading pre-defined profiles from /res/profiles"
level=INFO ts=2021-05-05T03:33:34.799961866Z app=device-virtual source=profiles.go:73 msg="Profile Random-Binary-Device exists, using the existing one"
level=INFO ts=2021-05-05T03:33:34.800887088Z app=device-virtual source=profiles.go:73 msg="Profile Random-Boolean-Device exists, using the existing one"
level=INFO ts=2021-05-05T03:33:34.801814469Z app=device-virtual source=profiles.go:73 msg="Profile Random-Float-Device exists, using the existing one"
level=INFO ts=2021-05-05T03:33:34.804896285Z app=device-virtual source=profiles.go:73 msg="Profile Random-Integer-Device exists, using the existing one"
level=INFO ts=2021-05-05T03:33:34.810810385Z app=device-virtual source=profiles.go:73 msg="Profile Random-UnsignedInteger-Device exists, using the existing one"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x8b7816]

goroutine 1 [running]:
github.com/edgexfoundry/device-sdk-go/v2/internal/provision.LoadDevices(0xc0003680c0, 0xd, 0xc0002a7340, 0x0, 0x0)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/internal/provision/devices.go:48 +0x1d6
github.com/edgexfoundry/device-sdk-go/v2/pkg/service.(*Bootstrap).BootstrapHandler(0xc00021fcc0, 0xc6e2e0, 0xc0002aaa00, 0xc000286b10, 0xc01ca263abff8e5b, 0x2f2a02, 0xfef5c0, 0xdf8475800, 0x3b9aca00, 0xc0002a7340, ...)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/pkg/service/init.go:74 +0x43e
github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap.RunAndReturnWaitGroup(0xc6e2e0, 0xc0002aaa00, 0xc000294d90, 0xc743e0, 0xc0002ae420, 0xba0a4c, 0xe, 0xba0a84, 0xe, 0xc73540, ...)
 /go/pkg/mod/github.com/edgexfoundry/go-mod-bootstrap/v2@v2.0.0-dev.34/bootstrap/bootstrap.go:158 +0x59d
github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap.Run(0xc6e2e0, 0xc0002aaa00, 0xc000294d90, 0xc743e0, 0xc0002ae420, 0xba0a4c, 0xe, 0xba0a84, 0xe, 0xc73540, ...)
 /go/pkg/mod/github.com/edgexfoundry/go-mod-bootstrap/v2@v2.0.0-dev.34/bootstrap/bootstrap.go:184 +0x158
github.com/edgexfoundry/device-sdk-go/v2/pkg/service.Main(0xba0a4c, 0xe, 0xc4ff10, 0xc, 0xb3cf60, 0xc0000694a0, 0xc6e2e0, 0xc0002aaa00, 0xc000294d90, 0xc0002b4540)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/pkg/service/main.go:65 +0x7fb
github.com/edgexfoundry/device-sdk-go/v2/pkg/startup.Bootstrap(0xba0a4c, 0xe, 0xc4ff10, 0xc, 0xb3cf60, 0xc0000694a0)
 /go/pkg/mod/github.com/edgexfoundry/device-sdk-go/v2@v2.0.0-dev.45/pkg/startup/bootstrap.go:19 +0x10e
main.main()
 /device-virtual-go/cmd/main.go:22 +0x72
```

**Comment by cloudxxx8** [comment]
fixed by #196
